File: config/config-java/src/test/java/quan/config/quest/QuestTargetConfig.java
package quan.config.quest;

import java.util.*;
import com.alibaba.fastjson.*;
import quan.config.*;

/**
* QuestTargetConfig<br/>
* Created by 自动生成
*/
public class QuestTargetConfig extends Config {

    //ID
    protected final int id;

    //名字
    protected final String name;


    public QuestTargetConfig(JSONObject json) {
        super(json);

        this.id = json.getIntValue("id");
        this.name = json.getOrDefault("name", "").toString();
    }

    /**
     * ID
     */
    public final int getId() {
        return id;
    }

    /**
     * 名字
     */
    public final String getName() {
        return name;
    }


    @Override
    protected QuestTargetConfig create(JSONObject json) {
        return new QuestTargetConfig(json);
    }

    @Override
    public String toString() {
        return "QuestTargetConfig{" +
                "id=" + id +
                ",name='" + name + '\'' +
                '}';

    }


    private volatile static List<QuestTargetConfig> configs = new ArrayList<>();

    //ID
    private volatile static Map<Integer, QuestTargetConfig> idConfigs = new HashMap<>();

    public static List<QuestTargetConfig> getConfigs() {
        return configs;
    }

    public static Map<Integer, QuestTargetConfig> getIdConfigs() {
        return idConfigs;
    }

    public static QuestTargetConfig getById(int id) {
        return idConfigs.get(id);
    }


    @SuppressWarnings({"unchecked"})
    public static List<String> index(List<QuestTargetConfig> configs) {
        Map<Integer, QuestTargetConfig> idConfigs = new HashMap<>();

        List<String> errors = new ArrayList<>();
        QuestTargetConfig oldConfig;

        for (QuestTargetConfig config : configs) {
            oldConfig = idConfigs.put(config.id, config);
            if (oldConfig != null) {
                String repeatedConfigs = config.getClass().getSimpleName();
                if (oldConfig.getClass() != config.getClass()) {
                    repeatedConfigs += "," + oldConfig.getClass().getSimpleName();
                }
                errors.add(String.format("配置[%s]有重复数据[%s = %s]", repeatedConfigs, "id", config.id));
            }
        }

        QuestTargetConfig.configs = Collections.unmodifiableList(configs);
        QuestTargetConfig.idConfigs = unmodifiableMap(idConfigs);

        return errors;
    }

}

File: config/ConfigCS/Test/Quest/QuestTargetConfig.cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using Newtonsoft.Json.Linq;
using ConfigCS;

namespace ConfigCS.Test.Quest
{
    /// <summary>
	/// QuestTargetConfig<br/>
	/// Created by 自动生成
	/// </summary>
    public class QuestTargetConfig : Config
    {

        /// <summary>
        /// ID
        /// </summary>
        public readonly int Id;

        /// <summary>
        /// 名字
        /// </summary>
        public readonly string Name;


        public QuestTargetConfig(JObject json): base(json)
        {
            Id = json["id"]?.Value<int>()?? default;
            Name = json["name"]?.Value<string>()?? "";
        }

        protected override Config Create(JObject json) 
        {
            return new QuestTargetConfig(json);
        }


        public override string ToString()
        {
            return "QuestTargetConfig{" +
                   "Id=" + Id +
                   ",Name='" + Name + '\'' +
                   '}';
        }


        private static volatile IList<QuestTargetConfig> _configs = new List<QuestTargetConfig>();

        /// <summary>
        /// ID
        /// </summary>
        private static volatile IDictionary<int, QuestTargetConfig> _idConfigs = new Dictionary<int, QuestTargetConfig>();

        public static IList<QuestTargetConfig> GetConfigs() 
        {
            return _configs;
        }

        public static IDictionary<int, QuestTargetConfig> GetIdConfigs() 
        {
            return _idConfigs;
        }

        public static QuestTargetConfig GetById(int id)
        {
            return _idConfigs.ContainsKey(id) ? _idConfigs[id] : null;
        }


        public static void Index(List<QuestTargetConfig> configs)
        {
            var idConfigs = new Dictionary<int, QuestTargetConfig>();

            foreach (var config in configs)
            {
                idConfigs[config.Id] = config;

            }

            _configs = configs.ToImmutableList();
            _idConfigs = idConfigs.ToImmutableDictionary();

        }
    }
}